apps/chat/conversation.py

```diff
@@ -39,9 +43,13 @@ def __init__(
         if experiment_session and experiment_session.experiment.tools_enabled:
             self.executer = AgentExecuter(llm=llm, memory=memory, experiment_session=experiment_session)
             # Insert the messages here
+            current_date_message = SystemMessagePromptTemplate.from_template("{current_date}")
             self.executer.agent.prompt = OpenAIFunctionsAgent.create_prompt(
                 system_message=prompt_to_use,
-                extra_prompt_messages=[MessagesPlaceholder(variable_name="history")],
+                extra_prompt_messages=[
+                    MessagesPlaceholder(variable_name="history"),
+                    current_date_message.format(current_date=str(current_datetime)),
```

Comment:
How does this work with history? Do we end up with multiple 'current_dates' in the chat history?

Reply:
Good question. Only the content of the user message / bot response get's saved to the chat history. See [this](https://github.com/dimagi/open-chat-studio/blob/95dff42ed292a391df3f420b77144aa4889846a6/apps/chat/bots.py#L118-L127).